grading/main_solution_runner.cpp

```diff
@@ -0,0 +1,140 @@
+#include <unistd.h>
```

Comment:
Let's get rid of this file. Instead, can we merge it with ```main_runner.cpp``` and provide a new argument ```--generate_solution```, which specifies that we are running ```main_runner``` in "generate mode." That way, we only have to maintain one script.